### ISSUE TITLE
HELM3 : values.yaml is re-ordered and comments stripped in edit window

### ISSUE THREAD

**Issue by morstad** [openshift/console#4736]
The values.yaml content displayed on install is re-ordered with comments stripped.  Removal of comments and re-ordering is concerning for following reasons  :  
- global and / or common parameters which need to be updated by user aren't displayed at top
- comments to guide users input are removed (thisincludes license acceptance text and pointer to README.MD externally hosted as not displayed during install flow in 4.4)
- parameter order doesn't follow install documentation

Example edit screen : 
![image](https://user-images.githubusercontent.com/24810680/76712960-39207e80-66eb-11ea-8f6c-b0f366c2a20a.png)

Source values.yaml : 
```
# For additional details regarding install parameters see : https://ibm.biz/oms-pro-install

global:
  # Change license to true to indicate have read and agree to license agreement : ibm/biz/oms-license
  licenseAccept: false
  license_store_call_center: true
  image:
    repository: ""
  appSecret: ""
...
```

@sbose78 @siamaksade @pedjak

**Comment by sbose78** [comment]
Thanks for logging this @morstad . The comments are lost because the deserialized values.yaml ignores comments. 

We are looking at the feasibility of pulling in the raw file.

**Comment by sbose78** [comment]
Based on the slack conversation with @pedjak , we need to check if https://github.com/helm/helm/blob/master/pkg/chart/chart.go#L36 can be used to populate the raw  file, in the response.

**Comment by pedjak** [comment]
> The comments/order are lost because the deserialized values.yaml ignores comments/order.

Here is the line where `values.yaml` gets parsed: https://github.com/helm/helm/blob/master/pkg/chart/loader/load.go#L97

**Comment by sbose78** [comment]
Suggestion: Provide another file called `values-{something}.yaml` that would be a symlink to values.yaml . If this file is available, the UI could render the yaml as is - thereby preserving the comments and order!

@pedjak & @rohitkrai03 
Please work this out :-)

**Comment by sbose78** [comment]
Let's call the file 
ordered-values.yaml

**Comment by sbose78** [comment]
@pedjak @akashshinde @rohitkrai03 Could we look for `ordered-values.yaml` to populate the yaml editor in the UI ?

> ordered-values.yaml
Hi @morstad Would this work ?

**Comment by pedjak** [comment]
@sbose78 the proposed name is fine (or any other). In any case @rohitkrai03 should need to look in Release response for that name under `files`. If there, it could be used for rendering.

**Comment by morstad** [comment]
@sbose78 orderd-values.yaml works - have instructed teams to add

**Comment by sbose78** [comment]
tracked internally https://issues.redhat.com/browse/ODC-3485

**Comment by pedjak** [comment]
For the record, `helm package` command puts into tarbals `values.yaml` cleaned from any comments and with alphabetically sorted keys. Thus, adding an extra `value.yaml` under another name is the only way to solve this, patching Helm would not help, because all charts in all repos in the wild contain only cleaned up `values.yaml`

**Comment by spadgett** [comment]
Fixed by #5037
/close
